code/game/objects/items/stunbaton.dm

```diff
@@ -194,26 +214,32 @@
 			L.visible_message("<span class='warning'>[user] has prodded [L] with [src]. Luckily it was out of charge.</span>", \
 							"<span class='warning'>[user] has prodded you with [src]. Luckily it was out of charge.</span>")
 			return FALSE
-		stunpwr *= round(stuncharge/hitcost, 0.1)
+		final_stamina_loss_amount *= round(stuncharge/hitcost, 0.1)
 
 	if(user && !user.UseStaminaBuffer(getweight(user, STAM_COST_BATON_MOB_MULT), warn = TRUE))
 		return FALSE
 
-	if(!disarming)
-		if(knockdown)
-			L.DefaultCombatKnockdown(50, override_stamdmg = 0)		//knockdown
-		L.adjustStaminaLoss(stunpwr)
-	else
-		L.drop_all_held_items()					//no knockdown/stamina damage, instead disarm.
-
-	L.apply_effect(EFFECT_STUTTER, stamforce)
+	if(shoving && COOLDOWN_FINISHED(src, shove_cooldown) && !HAS_TRAIT(L, TRAIT_IWASBATONED)) //Rightclicking applies a knockdown, but only once every couple of seconds, based on the cooldown_duration var. If they were recently knocked down, they can't be knocked down again by a baton.
+		L.DefaultCombatKnockdown(50, override_stamdmg = 0)
+		L.apply_status_effect(STATUS_EFFECT_TASED_WEAK, status_duration) //Even if they shove themselves up, they're still slowed.
+		L.apply_status_effect(STATUS_EFFECT_OFF_BALANCE, status_duration) //They're very likely to drop items if shoved briefly after a knockdown.
+		shoved = TRUE
+		COOLDOWN_START(src, shove_cooldown, cooldown_duration)
+		ADD_TRAIT(L, TRAIT_IWASBATONED, STATUS_EFFECT_TRAIT) //Prevents swapping to a new baton to avoid the cooldown by just acquiring more batons
+		addtimer(TRAIT_CALLBACK_REMOVE(L, TRAIT_IWASBATONED, STATUS_EFFECT_TRAIT), cooldown_duration)
+		playsound(loc, 'sound/weapons/zapbang.ogg', 50, 1, -1)
+	else //If we cannot/don't knock down the target, we apply a stagger, which keeps them from just running off
+		L.apply_status_effect(STATUS_EFFECT_STAGGERED, status_duration)
+
+	L.apply_damage (final_stamina_loss_amount, STAMINA, BODY_ZONE_CHEST, zap_block)
+	L.apply_effect(EFFECT_STUTTER, stamina_loss_amount)
 	SEND_SIGNAL(L, COMSIG_LIVING_MINOR_SHOCK)
 	if(user)
 		L.lastattacker = user.real_name
 		L.lastattackerckey = user.ckey
-		L.visible_message("<span class='danger'>[user] has [disarming? "disarmed" : "stunned"] [L] with [src]!</span>", \
-								"<span class='userdanger'>[user] has [disarming? "disarmed" : "stunned"] you with [src]!</span>")
-		log_combat(user, L, disarming? "disarmed" : "stunned")
+		L.visible_message("<span class='danger'>[user] has [shoved ? "knocked to the floor" : "stunned"] [L] with [src]!</span>", \
```

Comment:
```suggestion
		L.visible_message("<span class='danger'>[user] has [shoved ? "knocked [L] to the floor" : "stunned"] with [src]!</span>", \
```
>Redshirt McBeat has knocked to the floor Clown McShitbrain with the stunbaton!

Reply:
in hindsight this probably wouldn't work because brackets
🤔
double hindsight this is probably why it's fucked

Reply:
I'll try rephrasing it best I can